jvector-base/src/main/java/io/github/jbellis/jvector/quantization/PQLayout.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.jbellis.jvector.quantization;
+
+/**
+ * Chunk Dimensions and Layout
+ * This is emulative of modern Java records, but keeps to J11 standards.
+ * This class consolidates the layout calculations for PQ data into one place
+ */
+public class PQLayout {
```

Comment:
Instead of a public class in a separate file, should this not be a private subclass of PQVectors? They are coupled anyway... We can expose it for testing with the VisibleForTesting tag if needed.